sdk/core/Azure.Core/src/ClientOptions.cs

```diff
@@ -51,11 +59,12 @@ internal ClientOptions(ClientOptions? clientOptions)
             }
             else
             {
-                // Intentionally leaving this null. The only consumer of this branch is
-                // DefaultAzureCredential that would re-assign the value
-                _transport = null!;
-                Diagnostics = new DiagnosticsOptions();
-                Retry = new RetryOptions();
+                // Implementation Note: this code must use the copy constructors on DiagnosticsOptions and RetryOptions specifying
```

Comment:
Is this relevant? Do we even have the default constructors anymore?

Reply:
Yes this PR adds protected default constructors to `DiagnosticOptions` and `RetryOptions`.

Reply:
Ah I see. So the issue is that the default constructors for DiagnosticOptions/RetryOptions also use the same property that would invoke this path when being initialized.